File: java_sdk/src/main/java/io/swagger/client/api/VideoApi.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
package io.swagger.client.api;

import business_api.ApiException;
import business_api.SDKException;
import business_api.SDKExceptionForEvent;
import business_api.Response;
import business_api.ApiClient;
import business_api.Configuration;
import business_api.Pair;

import javax.ws.rs.core.GenericType;

import business_api_client.IdentityList;
import business_api_client.InlineResponse200;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-11-29T05:21:11.995104+05:30[Asia/Kolkata]")public class VideoApi {
  private ApiClient apiClient;

  public VideoApi() {
    this(Configuration.getDefaultApiClient());
  }

  public VideoApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  /**
   * Get GMV Max video info [Smart Plus GMV Max Video Get](https://business-api.tiktok.com/portal/docs?id&#x3D;1822001168512129)
   * 
   * @param advertiserId  (required)
   * @param storeId  (required)
   * @param storeAuthorizedBcId  (required)
   * @param accessToken Authorized access token. For details, see [Authentication](https://ads.tiktok.com/marketing_api/docs?id&#x3D;1738373164380162). (required)
   * @param spuIdList  (optional)
   * @param identityList  (optional)
   * @param needAuthCodeVideo  (optional)
   * @param page  (optional, default to 1)
   * @param pageSize  (optional, default to 10)
   * @param customPostsEligible  (optional)
   * @param sortField  (optional)
   * @param sortType  (optional, default to DESC)
   * @param keyword  (optional)
   * @return InlineResponse200
   * @throws ApiException if fails to make API call
   */
  public Response gmvMaxVideoGet(String advertiserId, String storeId, String storeAuthorizedBcId, String accessToken, List<String> spuIdList, List<IdentityList> identityList, Boolean needAuthCodeVideo, Long page, Long pageSize, Boolean customPostsEligible, String sortField, String sortType, String keyword) throws ApiException, SDKException, SDKExceptionForEvent {
    Object localVarPostBody = null;
    // verify the required parameter 'advertiserId' is set
    if (advertiserId == null) {
      throw new ApiException(400, "Missing the required parameter 'advertiserId' when calling gmvMaxVideoGet");
    }
    // verify the required parameter 'storeId' is set
    if (storeId == null) {
      throw new ApiException(400, "Missing the required parameter 'storeId' when calling gmvMaxVideoGet");
    }
    // verify the required parameter 'storeAuthorizedBcId' is set
    if (storeAuthorizedBcId == null) {
      throw new ApiException(400, "Missing the required parameter 'storeAuthorizedBcId' when calling gmvMaxVideoGet");
    }
    // verify the required parameter 'accessToken' is set
    if (accessToken == null) {
      throw new ApiException(400, "Missing the required parameter 'accessToken' when calling gmvMaxVideoGet");
    }
    // create path and map variables
    String localVarPath = "/open_api/v1.3/gmv_max/video/get/";

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "advertiser_id", advertiserId));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "store_id", storeId));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "store_authorized_bc_id", storeAuthorizedBcId));
    localVarQueryParams.addAll(apiClient.parameterToPairs("multi", "spu_id_list", spuIdList));
    localVarQueryParams.addAll(apiClient.parameterToPairs("multi", "identity_list", identityList));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "need_auth_code_video", needAuthCodeVideo));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "page", page));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "page_size", pageSize));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "custom_posts_eligible", customPostsEligible));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "sort_field", sortField));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "sort_type", sortType));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "keyword", keyword));

    if (accessToken != null)
      localVarHeaderParams.put("Access-Token", apiClient.parameterToString(accessToken));

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    GenericType<InlineResponse200> localVarReturnType = new GenericType<InlineResponse200>() {};
    InlineResponse200 response = apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
      if(response.getCode() != 0)
    {
      if (localVarPath.startsWith("/pixel")) {
          throw new SDKExceptionForEvent(response.getRequestId(), response.getCode(), response.getMessage(), response.getData());
      }
      throw new SDKException(response.getRequestId(), response.getCode(), response.getMessage());
    }
      Response return_response = new Response();
    return_response.setData(response.getData());
    return_response.setRequestId(response.getRequestId());

    return return_response;
  }
}

File: js_sdk/src/api/VideoApi.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from "../ApiClient.js";
import {IdentityList} from '../model/IdentityList.js';
import {InlineResponse200} from '../model/InlineResponse200.js';

/**
* Video service.
* @module api/VideoApi
* @version 1.2.1
*/
export class VideoApi {

    /**
    * Constructs a new VideoApi. 
    * @alias module:api/VideoApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instanc
    e} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }

    /**
     * Callback function to receive the result of the gmvMaxVideoGet operation.
     * @callback moduleapi/VideoApi~gmvMaxVideoGetCallback
     * @param {String} error Error message, if any.
     * @param {module:model/InlineResponse200{ data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get GMV Max video info [Smart Plus GMV Max Video Get](https://business-api.tiktok.com/portal/docs?id&#x3D;1822001168512129)
     * @param {String} advertiser_id 
     * @param {String} store_id 
     * @param {String} store_authorized_bc_id 
     * @param {String} Access_Token Authorized access token. For details, see [Authentication](https://ads.tiktok.com/marketing_api/docs?id&#x3D;1738373164380162).
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.spu_id_list 
     * @param {Array.<module:model/IdentityList>} opts.identity_list 
     * @param {Boolean} opts.need_auth_code_video 
     * @param {Number} opts.page  (default to <.>)
     * @param {Number} opts.page_size  (default to <.>)
     * @param {Boolean} opts.custom_posts_eligible 
     * @param {String} opts.sort_field 
     * @param {String} opts.sort_type  (default to <.>)
     * @param {String} opts.keyword 
     * @param {module:api/VideoApi~gmvMaxVideoGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link <&vendorExtensions.x-jsdoc-type>}
     */
    gmvMaxVideoGet(advertiser_id, store_id, store_authorized_bc_id, Access_Token, opts, callback) {
      opts = opts || {};
      let postBody = null;
      // verify the required parameter 'advertiser_id' is set
      if (advertiser_id === undefined || advertiser_id === null) {
        throw new Error("Missing the required parameter 'advertiser_id' when calling gmvMaxVideoGet");
      }
      // verify the required parameter 'store_id' is set
      if (store_id === undefined || store_id === null) {
        throw new Error("Missing the required parameter 'store_id' when calling gmvMaxVideoGet");
      }
      // verify the required parameter 'store_authorized_bc_id' is set
      if (store_authorized_bc_id === undefined || store_authorized_bc_id === null) {
        throw new Error("Missing the required parameter 'store_authorized_bc_id' when calling gmvMaxVideoGet");
      }
      // verify the required parameter 'Access_Token' is set
      if (Access_Token === undefined || Access_Token === null) {
        throw new Error("Missing the required parameter 'Access_Token' when calling gmvMaxVideoGet");
      }

      let pathParams = {
        
      };
      let queryParams = {
        'advertiser_id': advertiser_id,'store_id': store_id,'store_authorized_bc_id': store_authorized_bc_id,'spu_id_list': this.apiClient.buildCollectionParam(opts['spu_id_list'], 'multi'),'identity_list': this.apiClient.buildCollectionParam(opts['identity_list'], 'multi'),'need_auth_code_video': opts['need_auth_code_video'],'page': opts['page'],'page_size': opts['page_size'],'custom_posts_eligible': opts['custom_posts_eligible'],'sort_field': opts['sort_field'],'sort_type': opts['sort_type'],'keyword': opts['keyword']
      };
      let headerParams = {
        'Access-Token': Access_Token
      };
      let formParams = {
        
      };

      let authNames = [];
      let contentTypes = [];
      let accepts = ['application/json'];
      let returnType = InlineResponse200;

      return this.apiClient.callApi(
        '/open_api/v1.3/gmv_max/video/get/', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

}